tools/src/icon4py/tools/py2fgen/_generator.py

```diff
@@ -6,15 +6,32 @@
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
 
+import importlib
 import logging
 from pathlib import Path
+from typing import Callable
 
 import cffi
 
-from icon4py.tools.common.logger import setup_logger
+from icon4py.tools.py2fgen import _codegen
 
 
-logger = setup_logger(__name__)
+def get_cffi_description(
+    module_name: str, functions: list[str], plugin_name: str
+) -> _codegen.CffiPlugin:
+    module = importlib.import_module(module_name)
+    parsed_functions = [_get_function_descriptor(getattr(module, f)) for f in functions]
```

Comment:
instead of passing a list of functions, we can inspect which function is marked to export and export all of them

Reply:
I added this as future improvements to the readme